### ISSUE TITLE
JS and CLI API discussion

### ISSUE THREAD

**Issue by bclinkinbeard** [idyll-lang/idyll#40]
The current API looks like this in JS: `idyll(inputPath, opts, cb)` where `inputPath` is the absolute path to your `.idl` file, `opts` is an options object, and `cb` is a callback function that will receive the build artifacts once #39 lands.

The CLI is of course very similar, since it calls the JS API, looking something like this: `idyll index.idl --layout centered --css styles.css --build`, where `inputPath` is the first argument and everything else is translated to a property on the options object.

Of note:
- the CLI accepts relative paths 
- there is no analog to the callback when using the CLI

Resolution of `inputPath` should definitely move into the JS part so `inputPath` can be a relative path regardless of how Idyll is being used.

**Questions**
- Should the CLI return a stream or something to support piping into something else? Seems low priority.
- How should we support passing in Idyll source, rather than forcing the reading of a `.idl` file from disk?

One approach that seems fairly simple would be to support the omission of `inputPath` entirely, and in that scenario look for a `inputSrc` (or similar) property on `opts`. This is similar to how Browserify allows you to [specify `opts.entries`](https://github.com/substack/node-browserify#browserifyfiles--opts) instead of providing `files` as the first argument to the `browserify()` function.

Thoughts @mathisonian @rreusser et al?

**Comment by mathisonian** [comment]
I think updating it to something like 

```
var inputOptions = {
   inputString: ''
};

idyll(inputOptions, cb);
```

makes sense. I'm also in favor of moving all the file writing logic into the `idyll` bin, and out of the idyll javascript command itself. 

It doesn't seem necessary for the CLI to return a stream at this point. I think we should encourage people to use the API directly in that case.

**Comment by bclinkinbeard** [comment]
@mathisonian you mean make the `idyll()` function ALWAYS take an options object as the first parameter, and an optional callback as the second? And the options object would need to have either `inputPath` or `inputString` set?

Would that mean the CLI would look like `idyll --inputPath index.idl`?

**Comment by mathisonian** [comment]
Yeah, in that case it would always have options as the first argument. 

I'd leave the CLI as is - `idyll index.idl` - since that is a bit more in line with how other CLI tools work and seems more intuitive. I don't see it as an issue that the command line tool is slightly different than the javascript api.

**Comment by bclinkinbeard** [comment]
Cool, works for me
